### DISCUSSION TITLE
.WSL test returns false for WSL Debian

### DISCUSSION THREAD

**Discussion by bozho** [JanDeDobbeleer/oh-my-posh#4150] [Troubleshoot]
Hi,

I've noticed that `.WSL` test apparently returns `false` in my WSL Debian instance - the `os` segment doesn't show `WSL at` bit. The test works fine on a freshly installed Ubuntu 22.04 WSL instance.

`oh-my-posh` version on Debian was 18.2.0, just upgraded to 18.3.3. Ubuntu runs 18.3.2. I've tested with the official `violet` theme.

I know the test used to work on my Debian WSL instance, so I'm trying to figure out what had changed.

I'm not very familiar with Go, but I've looked at `src/platform/shell_unix.go`, `IsWsl` function (is that the function used by the `.WSL` test?). It looks like the function first tests for the existence of `IS_WSL` env. variable and if it doesn't find it, it looks for existence of the `wslpath` command.

I've tried exporting `IS_WSL=true` in my Debian shell, there was no change. `wslpath` command is available in Debian.

Any ideas on how to debug this?

Thank you!

**Comment by JanDeDobbeleer** [comment]
@bozho you can use oh-my-posh debug to see what logic oh-my-posh executes. It will report the values retrieved.

**Comment by bozho** [comment]
@JanDeDobbeleer Yup, `IsWsl` returns `false` on my Debian WSL:
```
[DEBUG] 11:50:34.175 shell_unix.go:IsWsl:35 → false
[TRACE] 11:50:34.175 shell_unix.go:IsWsl() - 3.4µs
[TRACE] 11:50:34.175 shell_unix.go:IsWsl2() - 6.8µs
[DEBUG] 11:50:34.177 shell_unix.go:IsWsl:35 → false
[TRACE] 11:50:34.177 shell_unix.go:IsWsl() - 4.4µs
```

Returns `true` on the Ubuntu WSL.

**Reply by JanDeDobbeleer** [reply]
Somewhere there should be a mention of `wslpath` in the logs, unless the cache is hit. The problem is that you'll never see this information as the cache will run before you can run debug. Maybe having debug ignore the cache would be a good idea.

**Reply by bozho** [reply]
`oh-my-posh debug --plain | grep -i wsl` only returns `IsWsl` calls and `[DEBUG] 13:42:05.725 shell.go:TemplateCache:810 → environment`, which contains `WSL` strings (`WSL_DISTRO_NAME` and `WSLENV`)

**Reply by bozho** [reply]
What information does cache actually cache that would cause this?

I don't know if it helps, but I remember that while this was working, SSHing into my Debian WSL instance would not show `WSL at` on the `os` segment.

Is `is_wsl` value cached early as `false` for some reason and then used in `IsWsl`?

**Reply by bozho** [reply]
Also doesn't know if it matters, but I'm sharing my home directory between Windows and Debian WSL (not the case with the Ubuntu WSL instance) and also use `oh-my-posh` in PowerShell.

Does the cache cache anything to the disk?

**Reply by JanDeDobbeleer** [reply]
The cache saves the presence of `wslpath`. If that executable exists, we set IsWSL to true.

**Comment by bozho** [comment]
Forgot to mention, both distros run on WSL 1
